pennylane/templates/broadcast.py

```diff
@@ -35,21 +37,20 @@ def wires_ring(wires):
     if len(wires) in [0, 1]:
         return []
     elif len(wires) == 2:
-        # exception: for 2 wires ring is set equal to chain,
-        # to avoid duplication of gate
-        return [[wires[0], wires[1]]]
+        # deviation from the rule: for 2 wires ring is set equal to chain,
+        # to avoid duplication of single gate
+        return [wires.subset([0, 1])]
     else:
-        ring = list(wires) + list(wires[0:1])
-        sequence = [[ring[i], ring[i + 1]] for i in range(len(wires))]
+        sequence = [wires.subset([i, i + 1], periodic_boundary=True) for i in range(len(wires))]
```

Comment:
So just for my understanding: here we have `periodic_boundary=True` such that for `i==len(wires)` we have `i==0` instead of an error for indexing out, correct?

A small example could be nice for `periodic_boundary=True` in the docstring for `Wires.subset`.

Reply:
Good idea!